test/functional/lua/vim_spec.lua

```diff
@@ -226,11 +226,8 @@ describe('lua stdlib', function()
     eq(false, fn.luaeval('vim.startswith("123", "2")'))
     eq(false, fn.luaeval('vim.startswith("123", "1234")'))
 
-    matches(
-      'prefix: expected string, got nil',
-      pcall_err(exec_lua, 'return vim.startswith("123", nil)')
-    )
-    matches('s: expected string, got nil', pcall_err(exec_lua, 'return vim.startswith(nil, "123")'))
+    matches('expected string, got nil', pcall_err(exec_lua, 'return vim.startswith("123", nil)'))
+    matches('expected string, got nil', pcall_err(exec_lua, 'return vim.startswith(nil, "123")'))
```

Comment:
since the string concatenation is guarded by a condition, is there any reason we can't mention the arg name in the message?
```suggestion
    matches('s: expected string, got nil', pcall_err(exec_lua, 'return vim.startswith(nil, "123")'))
```

Reply:
No reason other than we would have to pass the argument name and the argument itself to the function (an extra arg). Could be a bit tedious. vim.validate already does this though so it’s probably fine

Reply:
As long as it's still fast, a bit of tedium seems worth it for the UX improvement.

Reply:
Done here https://github.com/neovim/neovim/commit/8b88d99b2198cba251eb9b28717da523f7fb411f